pkg/controller/sparkapplication/controller.go

```diff
@@ -500,21 +532,33 @@ func (c *Controller) processSingleExecutorStateUpdate(update *executorStateUpdat
 
 func (c *Controller) updateSparkApplicationStatusWithRetries(
 	original *v1alpha1.SparkApplication,
-	updateFunc func(*v1alpha1.SparkApplicationStatus)) *v1alpha1.SparkApplication {
-	toUpdate := original.DeepCopy()
+	updateFunc func(*v1alpha1.SparkApplicationStatus),
+) *v1alpha1.SparkApplication {
 
 	var lastUpdateErr error
 	for i := 0; i < maximumUpdateRetries; i++ {
-		updated, err := c.tryUpdateStatus(original, toUpdate, updateFunc)
+		toUpdate := original.DeepCopy()
```

Comment:
You only need to make a deep copy once outside the loop. You can assign `toUpdate` to is fetched from the API server below as the one returned is a copy anyway.

Reply:
done

Reply:
We actually need this and I have reverted it back to make a deep copy inside the loop (https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/pull/227/commits/f600336b353287b0cd983148071c2bf9640b1135)

This is because for metrics,  we need the oldState of the CRD which we actually updated and since we run `updateFunc(&toUpdate.Status)` , `toUpdate` is no longer the oldState which was read from the API server nor is `original`. Let me know if you have any questions.